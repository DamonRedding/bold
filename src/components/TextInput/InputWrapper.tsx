import React, { CSSProperties } from 'react'

import { Theme, useStyles } from '../../styles'
import { Button, ButtonProps } from '../Button'
import { Icons } from '../Icon/generated/Icons'
import { Icon } from '../Icon/Icon'

export interface InputWrapperProps {
  icon?: Icons
  iconPosition?: 'left' | 'right'
  iconDisabled?: boolean
  clearVisible?: boolean
  onIconClick?: ButtonProps['onClick']
  children?: React.ReactNode
  onClear?(e: React.MouseEvent<HTMLButtonElement>): any
}

export function InputWrapper(props: InputWrapperProps) {
  const { children, icon, iconPosition, iconDisabled, onIconClick, clearVisible, onClear } = props
  const { classes, css } = useStyles(createStyles, props)

  const iconBoxClasses = css(
    classes.iconWrapper,
    iconPosition === 'left' && classes.iconLeft,
    iconPosition === 'right' && classes.iconRight
  )

  return (
    <div className={classes.wrapper}>
      {children}

      {clearVisible && (
        <span role='button' title='Limpar' tabIndex={-1} onClick={onClear} className={classes.clearButton}>
          <Icon size={1.5} icon='timesDefault' />
        </span>
      )}

      {icon && (
        <span className={iconBoxClasses}>
          {onIconClick ? (
            <Button
              size='small'
              skin='ghost'
              tabIndex={-1}
              onClick={onIconClick}
              style={classes.icon}
              disabled={iconDisabled}
            >
              <Icon icon={icon} />
            </Button>
          ) : (
            <Icon icon={icon} style={classes.icon} />
          )}
        </span>
      )}
    </div>
  )
}

InputWrapper.defaultProps = {
  iconPosition: 'right',
  iconDisabled: false,
  clearVisible: false,
  onClear: () => null,
} as Partial<InputWrapperProps>

const createStyles = (theme: Theme, { icon, iconPosition, clearVisible }: InputWrapperProps) => {
  const paddingRight =
    iconPosition === 'right'
      ? (clearVisible && icon && '4.5rem') || (clearVisible && '2rem') || (icon && '3rem')
      : clearVisible && '2rem'

  return {
    wrapper: {
      position: 'relative',
      input: {
        paddingLeft: iconPosition === 'left' && '3rem',
        paddingRight,
      },
    } as CSSProperties,
    clearButton: {
      position: 'absolute',
      right: icon && iconPosition === 'right' ? '2.5rem' : 1,
      background: 'transparent',
      border: 'none',
      cursor: 'pointer',
      height: '100%',
      display: 'inline-flex',
      justifyContent: 'center',
      alignItems: 'center',
      outline: 0,
      color: theme.pallete.text.secondary,
      padding: '1px 6px',
      '&:hover': {
        color: theme.pallete.status.danger.main,
      },
    } as CSSProperties,
    iconWrapper: {
      position: 'absolute',
      backgroundColor: theme.pallete.gray.c90,
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      padding: '0.25rem',
      width: '2.5rem',
      top: 1,
      bottom: 1,
    } as CSSProperties,
    iconLeft: {
      left: 1,
    } as CSSProperties,
    iconRight: {
      right: 1,
    } as CSSProperties,
    icon: {
      borderRadius: 'inherit',
      '&:focus': {
        boxShadow: 'none',
      },
    } as CSSProperties,
  }
}
