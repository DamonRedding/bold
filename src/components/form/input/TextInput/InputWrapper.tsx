import React from 'react'

import { Styles, withStyles, WithStylesProps } from '../../../../styles'
import { Button, ButtonProps } from '../../../elements/Button'
import { Icons } from '../../../elements/Icon/generated/Icons'
import { Icon } from '../../../elements/Icon/Icon'

export interface InputWrapperProps extends WithStylesProps {
    icon?: Icons
    iconPosition?: 'left' | 'right'
    iconDisabled?: boolean
    clearVisible?: boolean
    onIconClick?: ButtonProps['onClick']
    onClear?(e: React.MouseEvent<HTMLButtonElement>): any
}

@withStyles
export class InputWrapper extends React.PureComponent<InputWrapperProps> {

    static defaultProps: Partial<InputWrapperProps> = {
        iconPosition: 'right',
        iconDisabled: false,
        clearVisible: false,
        onClear: () => null,
    }

    render() {
        const {
            css, theme, children, icon, iconPosition, iconDisabled, onIconClick, clearVisible, onClear,
        } = this.props

        const paddingRight = iconPosition === 'right' ? (false
            || clearVisible && icon && '4.5rem'
            || clearVisible && '2rem'
            || icon && '3rem'
        ) : clearVisible && '2rem'

        const styles: Styles = {
            wrapper: {
                position: 'relative',
                'input': {
                    paddingLeft: iconPosition === 'left' && '3rem',
                    paddingRight,
                },
            },
            clearButton: {
                position: 'absolute',
                right: icon && iconPosition === 'right' ? '2.5rem' : 1,
                background: 'transparent',
                border: 'none',
                cursor: 'pointer',
                height: '100%',
                display: 'inline-flex',
                justifyContent: 'center',
                alignItems: 'center',
                outline: 0,
                color: theme.pallete.text.secondary,
                padding: '1px 6px',
                '&:hover': {
                    color: theme.pallete.status.danger.main,
                },
            },
            iconWrapper: {
                position: 'absolute',
                backgroundColor: theme.pallete.surface.background,
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center',
                padding: '0.25rem',
                width: '2.5rem',
                top: 1,
                bottom: 1,
            },
            iconLeft: { left: 1 },
            iconRight: { right: 1 },
            icon: {
                borderRadius: 'inherit',
                '&:focus': {
                    boxShadow: 'none',
                },
            },
        }

        const iconBoxClasses = [
            styles.iconWrapper,
            iconPosition === 'left' && styles.iconLeft,
            iconPosition === 'right' && styles.iconRight,
        ]

        return (
            <div className={css(styles.wrapper)}>
                {children}

                {clearVisible &&
                    <span
                        role='button'
                        title='Limpar'
                        tabIndex={-1}
                        onClick={onClear}
                        className={css(styles.clearButton)}
                    >
                        <Icon size={1.5} icon='timesDefault' />
                    </span>
                }

                {icon &&
                    <span className={css(iconBoxClasses)}>
                        {onIconClick ?
                            <Button
                                size='small'
                                skin='ghost'
                                tabIndex={-1}
                                onClick={onIconClick}
                                style={styles.icon}
                                disabled={iconDisabled}
                            >
                                <Icon icon={icon} />
                            </Button>
                            :
                            <Icon
                                icon={icon}
                                style={styles.icon}
                            />
                        }
                    </span>
                }
            </div>
        )
    }
}
