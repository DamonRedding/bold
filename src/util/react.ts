import { MutableRefObject, Ref } from 'react'

export function composeRefs<T>(...refs: Array<Ref<T>>) {
    return (value: T) => refs.forEach(ref => setRef(ref, value))
}

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}

/**
 * Returns the next sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getNextSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.nextElementSibling) {
        current = current.nextElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}

/**
 * Returns the previous sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getPreviousSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.previousElementSibling) {
        current = current.previousElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}
