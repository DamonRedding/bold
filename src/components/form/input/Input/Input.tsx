import * as React from 'react'

export interface InputController {
  getInput(): HTMLInputElement
  focus(): any
  blur(): any
}

export interface PublicInputProps {
  className?: string
  checked?: boolean
  disabled?: boolean
  id?: string
  maxLength?: number
  name?: string
  onBlur?: <T>(event?: React.FocusEvent<T>) => void
  onChange?: <T>(event: React.ChangeEvent<T> | any) => void
  onFocus?: <T>(event?: React.FocusEvent<T>) => void
  onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void
  placeholder?: string
  value?: any
  provideController?: (controller: InputController) => void
  autoFocus?: boolean
}

export interface InputProps extends PublicInputProps {
  type?: string
}

export class Input extends React.Component<InputProps, any> {
  private input: HTMLInputElement

  componentDidMount() {
    this.props.provideController && this.props.provideController({
      getInput: () => this.input,
      focus: () => this.input.focus(),
      blur: () => this.input.blur(),
    })
  }

  componentWillUnmount() {
    this.props.provideController && this.props.provideController(null)
  }

  render() {
    const { provideController, ...rest } = this.props

    return (
      <input ref={this.ref} {...rest} />
    )
  }

  private ref = (elem: HTMLInputElement) => {
    this.input = elem
  }

}
