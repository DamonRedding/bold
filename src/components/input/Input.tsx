import * as React from 'react'
import { EventOrValueHandler } from 'redux-form'
import * as classnames from 'classnames'
import { helpersClassnames, extractProps } from '../../util/Util'
import { ChangeEvent, FocusEvent } from 'react'

export interface PublicInputProps {
  className?: string
  checked?: boolean
  disabled?: boolean
  id?: string
  maxLength?: number
  name?: string
  onBlur?: EventOrValueHandler<FocusEvent<any>>
  onChange?: EventOrValueHandler<ChangeEvent<any>>
  onKeyPress?: (event: React.KeyboardEvent<HTMLInputElement>) => void
  placeholder?: string
  value?: any
}

export interface InputProps extends PublicInputProps {
  type: string
}

const propsToKeep = [
  'checked',
  'disabled',
  'id',
  'maxLength',
  'name',
  'onBlur',
  'onChange',
  'onKeyPress',
  'placeholder',
  'type',
  'value'
]

export class Input extends React.Component<InputProps, any> {
  public input: HTMLInputElement

  private ref = (elem: HTMLInputElement) => {
    this.input = elem
  }

  focus() {
    this.input.focus()
  }

  blur() {
    this.input.blur()
  }

  render() {
    const classes = classnames('input', this.props.className, helpersClassnames(this.props))

    return (
      <input ref={this.ref} {...extractProps(this.props, ...propsToKeep) } className={classes} />
    )
  }

}
