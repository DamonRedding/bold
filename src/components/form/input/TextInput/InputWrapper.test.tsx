import { render } from 'enzyme'
import * as React from 'react'

import { withTheme } from '../../../../test'
import { Input } from '../Input/Input'

import { InputWrapper } from './InputWrapper'

it('should render correctly', () => {
    const wrapper = render(withTheme(
        <InputWrapper clearVisible={true}>
            <Input />
        </InputWrapper>
    ))
    expect(wrapper).toMatchSnapshot()
})

it('should render correctly with button', () => {
    const wrapper = render(withTheme(
        <InputWrapper icon='search' clearVisible={true} onIconClick={jest.fn()}>
            <Input />
        </InputWrapper>
    ))
    expect(wrapper).toMatchSnapshot()
})

it('should render correctly with left position', () => {
    const wrapper = render(withTheme(
        <InputWrapper icon='search' clearVisible={true} iconPosition='left'>
            <Input />
        </InputWrapper>
    ))
    expect(wrapper).toMatchSnapshot()
})

it('should render correctly without clear icon', () => {
    const wrapper = render(withTheme(
        <InputWrapper icon='search' clearVisible={false}>
            <Input />
        </InputWrapper>
    ))
    expect(wrapper).toMatchSnapshot()
})

it('should render correctly without clear icon AND left position', () => {
    const wrapper = render(withTheme(
        <InputWrapper icon='search' clearVisible={false} iconPosition='left' >
            <Input />
        </InputWrapper>
    ))
    expect(wrapper).toMatchSnapshot()
})
