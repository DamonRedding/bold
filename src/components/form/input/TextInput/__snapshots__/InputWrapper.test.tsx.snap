// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.emotion-2 {
  position: relative;
}

.emotion-2 input {
  padding-right: 2rem;
}

.emotion-1 {
  position: absolute;
  right: 1px;
  background: transparent;
  border: none;
  cursor: pointer;
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: 0;
  color: #53595E;
  padding: 1px 6px;
}

.emotion-1:hover {
  color: #D83A2E;
}

.emotion-0 {
  fill: currentColor;
  font-size: 1.5rem;
}

<div>
  <div
    class="emotion-2"
  >
    <input />
    <span
      class="emotion-1"
      role="button"
      tabindex="-1"
      title="Limpar"
    >
      <svg
        aria-hidden="true"
        class="emotion-0"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M8.071 6.657l4.243 4.242 4.242-4.242c.464-.48.935-.48 1.415 0s.48.95 0 1.414l-4.243 4.243 4.243 4.242c.471.472.471.943 0 1.415-.472.471-.943.471-1.415 0l-4.242-4.243L8.07 17.97c-.46.482-.932.482-1.414 0-.482-.483-.482-.954 0-1.415l4.242-4.242L6.657 8.07c-.472-.47-.472-.942 0-1.414s.943-.472 1.414 0z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`should render correctly with button 1`] = `
.emotion-4 {
  position: absolute;
  background-color: #F2F2F7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.25rem;
  width: 2.5rem;
  top: 1px;
  bottom: 1px;
  right: 1px;
}

.emotion-0 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-5 {
  position: relative;
}

.emotion-5 input {
  padding-right: 4.5rem;
}

.emotion-1 {
  position: absolute;
  right: 2.5rem;
  background: transparent;
  border: none;
  cursor: pointer;
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: 0;
  color: #53595E;
  padding: 1px 6px;
}

.emotion-1:hover {
  color: #D83A2E;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.5rem;
  position: relative;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all .2s;
  transition: all .2s;
  font-weight: bold;
  cursor: pointer;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: 4px;
  color: #2D343B;
  font-size: 0.875rem;
  padding: calc(0.25rem - 1px) calc(0.5rem - 1px);
  border-radius: inherit;
}

.emotion-3 > span {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-3:not(:disabled):active {
  box-shadow: inset 0 2px 8px 0 rgba(0,0,0,0.1);
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0 0 0 2px #0A67D1;
}

.emotion-3:not(:disabled):hover {
  background-color: rgba(45,52,59,0.1);
}

.emotion-3:focus {
  box-shadow: none;
}

<div>
  <div
    class="emotion-5"
  >
    <input />
    <span
      class="emotion-1"
      role="button"
      tabindex="-1"
      title="Limpar"
    >
      <svg
        aria-hidden="true"
        class="emotion-0"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M8.071 6.657l4.243 4.242 4.242-4.242c.464-.48.935-.48 1.415 0s.48.95 0 1.414l-4.243 4.243 4.243 4.242c.471.472.471.943 0 1.415-.472.471-.943.471-1.415 0l-4.242-4.243L8.07 17.97c-.46.482-.932.482-1.414 0-.482-.483-.482-.954 0-1.415l4.242-4.242L6.657 8.07c-.472-.47-.472-.942 0-1.414s.943-.472 1.414 0z"
        />
      </svg>
    </span>
    <span
      class="emotion-4"
    >
      <button
        class="emotion-3"
        tabindex="-1"
        type="button"
      >
        <span>
          <svg
            aria-hidden="true"
            class="emotion-0"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M12.89 11.478l1.936 1.936c.943-.943 1.886-.943 2.828 0l3.829 3.829c.943.942.943 1.885 0 2.828l-1.414 1.414c-.943.943-1.886.943-2.829 0l-3.828-3.828c-.943-.943-.942-1.887.002-2.831-.001.001-.647-.644-1.937-1.936a6 6 0 1 1 1.412-1.412zm1.99 4.744l3.83 3.829 1.414-1.415-3.829-3.828-1.414 1.414zM8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8z"
            />
          </svg>
        </span>
      </button>
    </span>
  </div>
</div>
`;

exports[`should render correctly with left position 1`] = `
.emotion-1 {
  position: absolute;
  right: 1px;
  background: transparent;
  border: none;
  cursor: pointer;
  height: 100%;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: 0;
  color: #53595E;
  padding: 1px 6px;
}

.emotion-1:hover {
  color: #D83A2E;
}

.emotion-0 {
  fill: currentColor;
  font-size: 1.5rem;
}

.emotion-4 {
  position: relative;
}

.emotion-4 input {
  padding-left: 3rem;
  padding-right: 2rem;
}

.emotion-3 {
  position: absolute;
  background-color: #F2F2F7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.25rem;
  width: 2.5rem;
  top: 1px;
  bottom: 1px;
  left: 1px;
}

.emotion-2 {
  fill: currentColor;
  font-size: 1.5rem;
  border-radius: inherit;
}

.emotion-2:focus {
  box-shadow: none;
}

<div>
  <div
    class="emotion-4"
  >
    <input />
    <span
      class="emotion-1"
      role="button"
      tabindex="-1"
      title="Limpar"
    >
      <svg
        aria-hidden="true"
        class="emotion-0"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M8.071 6.657l4.243 4.242 4.242-4.242c.464-.48.935-.48 1.415 0s.48.95 0 1.414l-4.243 4.243 4.243 4.242c.471.472.471.943 0 1.415-.472.471-.943.471-1.415 0l-4.242-4.243L8.07 17.97c-.46.482-.932.482-1.414 0-.482-.483-.482-.954 0-1.415l4.242-4.242L6.657 8.07c-.472-.47-.472-.942 0-1.414s.943-.472 1.414 0z"
        />
      </svg>
    </span>
    <span
      class="emotion-3"
    >
      <svg
        aria-hidden="true"
        class="emotion-2"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12.89 11.478l1.936 1.936c.943-.943 1.886-.943 2.828 0l3.829 3.829c.943.942.943 1.885 0 2.828l-1.414 1.414c-.943.943-1.886.943-2.829 0l-3.828-3.828c-.943-.943-.942-1.887.002-2.831-.001.001-.647-.644-1.937-1.936a6 6 0 1 1 1.412-1.412zm1.99 4.744l3.83 3.829 1.414-1.415-3.829-3.828-1.414 1.414zM8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`should render correctly without clear icon 1`] = `
.emotion-1 {
  position: absolute;
  background-color: #F2F2F7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.25rem;
  width: 2.5rem;
  top: 1px;
  bottom: 1px;
  right: 1px;
}

.emotion-0 {
  fill: currentColor;
  font-size: 1.5rem;
  border-radius: inherit;
}

.emotion-0:focus {
  box-shadow: none;
}

.emotion-2 {
  position: relative;
}

.emotion-2 input {
  padding-right: 3rem;
}

<div>
  <div
    class="emotion-2"
  >
    <input />
    <span
      class="emotion-1"
    >
      <svg
        aria-hidden="true"
        class="emotion-0"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12.89 11.478l1.936 1.936c.943-.943 1.886-.943 2.828 0l3.829 3.829c.943.942.943 1.885 0 2.828l-1.414 1.414c-.943.943-1.886.943-2.829 0l-3.828-3.828c-.943-.943-.942-1.887.002-2.831-.001.001-.647-.644-1.937-1.936a6 6 0 1 1 1.412-1.412zm1.99 4.744l3.83 3.829 1.414-1.415-3.829-3.828-1.414 1.414zM8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8z"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`should render correctly without clear icon AND left position 1`] = `
.emotion-1 {
  position: absolute;
  background-color: #F2F2F7;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.25rem;
  width: 2.5rem;
  top: 1px;
  bottom: 1px;
  left: 1px;
}

.emotion-0 {
  fill: currentColor;
  font-size: 1.5rem;
  border-radius: inherit;
}

.emotion-0:focus {
  box-shadow: none;
}

.emotion-2 {
  position: relative;
}

.emotion-2 input {
  padding-left: 3rem;
}

<div>
  <div
    class="emotion-2"
  >
    <input />
    <span
      class="emotion-1"
    >
      <svg
        aria-hidden="true"
        class="emotion-0"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
      >
        <path
          d="M12.89 11.478l1.936 1.936c.943-.943 1.886-.943 2.828 0l3.829 3.829c.943.942.943 1.885 0 2.828l-1.414 1.414c-.943.943-1.886.943-2.829 0l-3.828-3.828c-.943-.943-.942-1.887.002-2.831-.001.001-.647-.644-1.937-1.936a6 6 0 1 1 1.412-1.412zm1.99 4.744l3.83 3.829 1.414-1.415-3.829-3.828-1.414 1.414zM8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8z"
        />
      </svg>
    </span>
  </div>
</div>
`;
